msu/hooks/skills/skill_container.nut

```diff
@@ -0,0 +1,381 @@
+::mods_hookNewObject("skills/skill_container", function(o) {
+	o.m.BeforeSkillExecutedTile <- null;
+	o.m.IsExecutingMoveSkill <- false;
+
+	local update = o.update;
+	o.update = function()
+	{
+		if (this.m.IsUpdating || !this.m.Actor.isAlive())
+		{
+			return;
+		}
+
+		foreach ( skill in this.m.Skills )
+		{
+			skill.softReset();
+		}
+
+		update();
+
+		foreach ( skill in this.m.Skills )
+		{
+			skill.executeScheduledChanges();
+		}
+	}
+
+	o.doOnFunction <- function( _function, _argsArray = null, _update = true, _aliveOnly = false )
+	{
+		if (_argsArray == null) _argsArray = [null];
+		else _argsArray.insert(0, null);
+
+		local wasUpdating = this.m.IsUpdating;
+		this.m.IsUpdating = true;
+		this.m.IsBusy = false;
+		this.m.BusyStack = 0;
+
+		foreach ( skill in this.m.Skills )
+		{
+			if (_aliveOnly && !this.m.Actor.isAlive())
+			{
+				break;
+			}
+
+			if (!skill.isGarbage())
+			{
+				_argsArray[0] = skill;
+				skill[_function].acall(_argsArray);
+			}
+		}
+
+		this.m.IsUpdating = wasUpdating;
+		
+		if (_update)
+		{
+			this.update();
+		}
+	}
+
+	o.doOnFunctionWhenAlive <- function( _function, _argsArray = null, _update = true )
+	{
+		this.doOnFunction(_function, _argsArray, _update, true);
+	}
+
+	o.buildProperties <- function( _function, _argsArray )
+	{
+		_argsArray.insert(0, null);
+		_argsArray.push(this.m.Actor.getCurrentProperties().getClone());
+
+		local wasUpdating = this.m.IsUpdating;
+		this.m.IsUpdating = true;
+
+		foreach (skill in this.m.Skills)
+		{
+			_argsArray[0] = skill;
+			skill[_function].acall(_argsArray);
```

Comment:
It doesn't, because call is basically
```squirrel
function call( _env, ... )
```
so if you pass it an array as ... it'll just consider that to be the first arg

Reply:
Pretty sure `call` doesn't accept `vargv`, it will instead treat _argsArray as an argument and pass it to _function.